### ISSUE TITLE
[FEA] String:: join_strings api - `na_rep` parameter doesn't work in specific scenario

### ISSUE THREAD

**Issue by galipremsagar** [rapidsai/cudf#4395]
**Describe the bug**
This issue is related to `string::join_strings`(not `string::concatenate`) api.

For the `.str.cat` API, if there is only one column as input and there is no `others` param, `string::join_strings` is invoked. In such case there seems to be two things(one bug and one feature request) with this api:
1. [Bug] If there is one string and rest all values are None's in a column then the api is respecting `na_rep` parameter, but when all the values in the input column are `None`, the api doesn't seem to respect what the value of `na_rep` is and returns `None` as a result.
2. [Feature request] If all the elements of the column are None, the result the underlying libcudf API returns None, is there any possibility to return empty string like pandas does? If it's not possible we can handle it in python layer aswell but since we might have to visit `na_rep` in `join_strings` as stated above just checking if this is feasible. 

Sample code below.

**Steps/Code to reproduce bug**
**Code for 1 [Bug]**
```python
>>> import cudf
>>> s = cudf.Series([None, None, 'abc'], dtype='str')
>>> s.str.cat(na_rep='3')
'33abc'
>>> s.to_pandas().str.cat(na_rep='3')
'33abc'

>>> s = cudf.Series([None, None, None], dtype='str')
>>> s.str.cat(na_rep='3')
>>> s.str.cat(na_rep='3') == None
True
>>> s.to_pandas().str.cat(na_rep='3')
'333'
```

**Code for 2 [Feature request]**
```python
>>> s = cudf.Series([None, None, None], dtype='str')
>>> s.str.cat()
>>> s.str.cat() == None
True
>>> s.to_pandas().str.cat()
''
```

**Expected behavior**
Expected behavior is to adhere `na_rep` even when all elements are `None`.

**Environment overview (please complete the following information)**
 - Environment location: Docker
 - Method of cuDF install: from source

**Environment details**
output of the `cudf/print_env.sh` script here: [env.txt](https://github.com/rapidsai/cudf/files/4312773/env.txt)

**Additional context**
None

**Comment by davidwendt** [comment]
I found the bug for (1) in the logic in `cudf::strings::join_strings()` but I don't see the logic for (2) in libcudf. Could it be in the cython/python layer?

**Comment by kkraus14** [comment]
@galipremsagar is this resolved since @davidwendt fixed (1)?

For (2) I think this makes sense to special case for Python. Where we can check if the output is `None` and then return `''` instead.

**Comment by galipremsagar** [comment]
@kkraus14, yes this is fixed with #4436 as I made similar changes in that PR to facilitate python side check for it.
